komorebi/consts.in

```diff
@@ -0,0 +1,4 @@
+package_name : str = @PACKAGE_NAME@
+package_version : str = @PACKAGE_VERSION@
+package_datadir : str = @PKGDATADIR@
+datadir : str = @DATADIR@
```

Comment:
I am not really a fan of generating this file... I think we should support multiple data directories instead of hardcoding one.

Also note, the standard way to export a version is in `module.__version__`.

Reply:
> support multiple data directories...

Do you mean support various places where the app can load wallpapers? We already added support for that, it has been reimplemented in [settings.py](https://github.com/Komorebi-Fork/komorebi/pull/63/files?file-filters%5B%5D=.build&file-filters%5B%5D=.py&file-filters%5B%5D=.vala#diff-e2e305757dab8b68fc13916733bd332608b4d560ef426add845e88d18b46451aR145)

Reply:
Then why do we need to hardcode a data directory?

Reply:
Because the official wallpapers are installed at `@PKGDATADIR@` _(aka `/usr/local/share/komorebi`)_. Plus, since we can't embed `.gresource` files into the binary *(since this is Python)*, the standard path to place them is this one too, so we need to know where was it installed to.

`@DATADIR@` _(aka `/usr/local/share`)_ is only used so we can access the `.desktop` file of Komorebi to implement autostart on boot functionality.

Reply:
Don't install them at `@PKGDATADIR@`, put them and `.gresource` next to the code and do `os.path.join(__file__, '..', 'something.gresource')`. `__file__` is the path of the file being run.

A `.desktop` file can be provided in the repo, users installing this via pip would not get it but packagers would install it.

The current approach is *fine*, but not compatible with the standard Python distribution. I think it would make sense to make this pip-installable, as it would simply the installation process for users.